sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSender.java

```diff
@@ -664,11 +674,11 @@ private void completeClose() {
      */
     private void handleError(Throwable error) {
         synchronized (pendingSendLock) {
-            final String logMessage = isDisposed.getAndSet(true)
-                ? "This was already disposed. Dropping error."
-                : String.format("Disposing of '%d' pending sends with error.", pendingSendsMap.size());
-            logger.verbose("connectionId[{}] entityPath[{}] linkName[{}] {}", handler.getConnectionId(), entityPath,
-                getLinkName(), logMessage);
+            if (isDisposed.getAndSet(true)) {
+                logger.verbose("This was already disposed. Dropping error.");
+            } else {
+                logger.verbose("Disposing of '{}' pending sends with error.", pendingSendsMap.size());
```

Comment:
`pendingSendsMap` is a concurrent hashmap and the size is computed each time. Maybe this can be added as key-value pair using the supplier to compute the size only if logging at verbose level is enabled.